File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if(nums.size() <= 1) return 1; 

        int maxJump = 0; 
        for(int i = 0; i < nums.size(); i++){
            if(i > maxJump) return false; 
            maxJump = max(maxJump, nums[i] + i); 
            if(maxJump >= nums.size()-1) return true; 
        }
        return true; 
    }
};

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        if(nums.length == 1) return true; 
        int maxJump = nums[0];  
        for(int i = 0; i <= maxJump; i++){
            maxJump = Math.max(maxJump, i + nums[i]); 
            if(maxJump >= nums.length-1) return true; 
        }
        return false; 
    }
}